languages/REXX.groovy

```diff
@@ -0,0 +1,235 @@
+@Field RepositoryClient repositoryClient
+
+println("** Building files mapped to ${this.class.getName()}.groovy script")
+
+// verify required build properties
+buildUtils.assertBuildProperties(props.REXX_requiredBuildProperties)
+
+// create language datasets
+def langQualifier = "REXX"
+buildUtils.createLanguageDatasets(langQualifier)
+
+// sort the build list based on build file rank if provided
+List<String> sortedList = buildUtils.sortBuildList(argMap.buildList, 'REXX_fileBuildRank')
+
+// iterate through build list
+sortedList.each { buildFile ->
+	println "*** Building file $buildFile"
+
+	// copy build file and dependency files to data sets
+	String rules = props.getFileProperty('REXX_resolutionRules', buildFile)
+	DependencyResolver dependencyResolver = buildUtils.createDependencyResolver(buildFile, rules)
+	buildUtils.copySourceFiles(buildFile, props.REXX_srcPDS, props.REXX_srcPDS, dependencyResolver)
+	// create mvs commands
+	LogicalFile logicalFile = dependencyResolver.getLogicalFile()
+	String member = CopyToPDS.createMemberName(buildFile)
+	File logFile = new File( props.userBuild ? "${props.buildOutDir}/${member}.log" : "${props.buildOutDir}/${member}.REXX.log")
+	if (logFile.exists())
+		logFile.delete()
+	MVSExec compile = createCompileCommand(buildFile, logicalFile, member, logFile)
+	File linkEditLogFile = new File( props.userBuild ? "${props.buildOutDir}/${member}.LinkEdit.log" : "${props.buildOutDir}/${member}.REXX.LinkEdit.log")
+	if (linkEditLogFile.exists())
+		linkEditLogFile.delete()
+	MVSExec linkEdit = createLinkEditCommand(buildFile, logicalFile, member, linkEditLogFile)
+
+
+	// execute mvs commands in a mvs job
+	MVSJob job = new MVSJob()
+	job.start()
+
+	// compile the cobol program
+	int rc = compile.execute()
+	int maxRC = props.getFileProperty('REXX_compileMaxRC', buildFile).toInteger()
+
+	boolean bindFlag = true
+
+	if (rc > maxRC) {
+		bindFlag = false
+		String errorMsg = "*! The compile return code ($rc) for $buildFile exceeded the maximum return code allowed ($maxRC)"
+		println(errorMsg)
+		props.error = "true"
+		buildUtils.updateBuildResult(errorMsg:errorMsg,logs:["${member}.log":logFile],client:getRepositoryClient())
+	}
+	else {
+		// if this program needs to be link edited . . .
+		String needsLinking = props.getFileProperty('REXX_linkEdit', buildFile)
+		if (needsLinking.toBoolean()) {
+			rc = linkEdit.execute()
+			maxRC = props.getFileProperty('REXX_linkEditMaxRC', buildFile).toInteger()
+
+			if (rc > maxRC) {
+				String errorMsg = "*! The link edit return code ($rc) for $buildFile exceeded the maximum return code allowed ($maxRC)"
+				println(errorMsg)
+				props.error = "true"
+				buildUtils.updateBuildResult(errorMsg:errorMsg,logs:["${member}.log":logFile],client:getRepositoryClient())
+			}
+			else {
+				if (!props.userBuild){
+					// only scan the load module if load module scanning turned on for file
```

Comment:
More a question ... can we scan a REXX LOAD. Can we have several CSECTs in the load?

Reply:
I've not tried to scan the loadmodule when it was created with CEXEC. If you use CEXEC, you probably want to disable the traditional linkedit step, so your loadmodule is not scanned. Scanning works when the OBJECT deck is generated and the link edit is done with the z/OS binder.